packages/vite/src/node/server/hmr.ts

```diff
@@ -50,7 +50,7 @@ export async function handleHMRUpdate(
   const isConfigDependency = config.configFileDependencies.some(
     (name) => file === path.resolve(name)
   )
-  const isEnv = config.inlineConfig.envFile !== false && file.endsWith('.env')
+  const isEnv = config.inlineConfig.envFile !== false && file.startsWith('.env')
```

Comment:
We could also use a more strict regex here 🤔

Reply:
I suppose `file === '.env' || file.startsWith('.env.')` might be slightly stricter, but this probably seems okay to me either way

Reply:
Yes, I think we should use your new proposed solution 👍